src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/IfError.txt

```diff
@@ -195,3 +198,24 @@ Time(11,30,59,0)
 // Error-DateTime
 >> IfError(Error({Kind: ErrorKind.Validation}), DateTime(2021, 12, 12, 17, 30, 0))
 DateTime(2021,12,12,17,30,0,0)
+
+>> IfError(1, {x:1}, {x:1}, {y:1}, 1)
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+// IfError can have mismatched arg types, but result can't be used.
+>> IfError(1/0, 1, {x:4})
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+// IfError accepts the void value. But then result of the expression can't be used. If(true, 1, {x:4}) => void value.
+>> IfError(1/0, 1, If(true, 1, {x:4}), 1)
+If(true, {test:1}, "Mismatched args (result of the expression can't be used).")
+
+// Errors are still returned.
+>> IfError(1, 2, 3/0, If(true, 1/0, {x:4}))
+Error({Kind:ErrorKind.Div0})
+
+>> IfError(1, a)
```

Comment:
Another interesting test would be if we have a Void expression inside IfError that cannot be returned - in this case the result should be non-Void. Something along the lines of

    >> IfError( If(true, 1, {a:1}), 1, 2 )
    2

    >> IfError( If(true, 1/0, {a:1}), 1, 2 )
    1

    >> IfError( 1, "hello", If(true,1/0,{a:1}), "great", "world")
    "great"

    >> IfError( 1, "one", If(true,1,{a:1}), "two", "three")
    "three"

Reply:
Added! Thanks for the tests. :)